perf/reader.go

```diff
@@ -131,6 +146,16 @@ func readRawSample(rd io.Reader, buf, sampleBuf []byte) ([]byte, error) {
 	return data, nil
 }
 
+func readRawEvent(rd io.Reader, header perfEventHeader) ([]byte, error) {
```

Comment:
Similar question to `RawSample`, why can't this reuse `readRawSample`?

This function should take a pre-allocated buffer like `readRawSample` so that reusing a record is zero allocation.

Reply:
yep, good idea

Reply:
actually the readRawSample is reading the bpf sample which is different from what we want to read for raw events - the whole perf event including the header.. I pushed new version with that and added comments for that